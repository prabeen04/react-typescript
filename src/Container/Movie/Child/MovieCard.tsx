import * as React from 'react';
import { useSpring, animated } from "react-spring";
import { IMovie } from "../MovieInterface";
const IMG_URL = 'http://image.tmdb.org/t/p/w185/'

interface IMovieCardProps {
    movie: IMovie
}
export default function MovieCard({ movie }: { movie: IMovie }): JSX.Element {
    const [flipped, setFlipped] = React.useState<boolean>(false)
    const { transform, opacity } = useSpring({
        opacity: flipped ? 1 : 0,
        transform: `perspective(600px) rotateX(${flipped ? 180 : 0}deg)`,
        config: { mass: 5, tension: 500, friction: 80 }
    })
    return (
        <div className='movie-card' data-test='MovieCard'>
            <p>{movie.title}</p>
            <div className='flex-container'>
                <div onClick={() => setFlipped(state => !state)}>
                    <animated.div className="c back" style={{ opacity: opacity.interpolate((o: any) => 1 - o), transform }} >
                        <img className='movie-poster' src={`${IMG_URL}${movie.poster_path}`} alt="" />
                    </animated.div>
                    <animated.div className="c front" style={{ opacity, transform: transform.interpolate(t => `${t} rotateX(180deg)`) }} >
                        <p className='movie-description' >{movie.overview} </p>
                    </animated.div>
                </div>

            </div>
            <div className="movie-stats">
                <div className="stats-column">
                    <i className="fas fa-calendar-day"></i>
                    <span>{movie.release_date}</span>
                </div>
                <div className="stats-column">
                    <i className="fas fa-heartbeat"></i>
                    <span>{movie.vote_count}</span>
                </div>
                <div className="stats-column">
                    <i className="fas fa-star"></i>
                    <span>{movie.vote_average}</span>
                </div>
            </div>

        </div>
    )
}
