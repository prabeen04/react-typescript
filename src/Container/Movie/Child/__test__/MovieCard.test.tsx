import * as React from 'react';
import { shallow, mount } from "enzyme";
import MovieCard from '../MovieCard';
import { IMovie } from "../../MovieInterface";

describe('<MovieCard/>', () => {
    const movieProp: IMovie = {
        adult: false,
        backdrop_path: "/h3KN24PrOheHVYs9ypuOIdFBEpX.jpg",
        genre_ids: [16, 10751, 12],
        id: 166428,
        original_language: "en",
        original_title: "How to Train Your Dragon: The Hidden World",
        overview: "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.",
        popularity: 470.427,
        poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
        release_date: "2019-01-03",
        title: "How to Train Your Dragon: The Hidden World",
        video: false,
        vote_average: 7.7,
        vote_count: 1230,
    }
    it('should match snapshot', () => {
        const wrapper = shallow(<MovieCard movie={movieProp} />)
        expect(wrapper).toMatchSnapshot()
    })
    
    it('should render', () => {
        const wrapper = shallow(<MovieCard movie={movieProp} />)
        const element = wrapper.find('[data-test="MovieCard"]')
        expect(element.length).toEqual(1)
    })
    it('should recieve movie props', () => {
        const wrapper = mount(<MovieCard movie={movieProp} />)
        expect(wrapper.prop('movie')).toEqual(movieProp)
    })
})
