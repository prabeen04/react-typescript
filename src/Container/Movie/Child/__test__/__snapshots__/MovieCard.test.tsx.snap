// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/h3KN24PrOheHVYs9ypuOIdFBEpX.jpg",
        "genre_ids": Array [
          16,
          10751,
          12,
        ],
        "id": 166428,
        "original_language": "en",
        "original_title": "How to Train Your Dragon: The Hidden World",
        "overview": "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.",
        "popularity": 470.427,
        "poster_path": "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
        "release_date": "2019-01-03",
        "title": "How to Train Your Dragon: The Hidden World",
        "video": false,
        "vote_average": 7.7,
        "vote_count": 1230,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          How to Train Your Dragon: The Hidden World
        </p>,
        <img
          alt=""
          src="http://image.tmdb.org/t/p/w185//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg"
        />,
      ],
      "className": "movie-card",
      "data-test": "MovieCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "How to Train Your Dragon: The Hidden World",
        },
        "ref": null,
        "rendered": "How to Train Your Dragon: The Hidden World",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "http://image.tmdb.org/t/p/w185//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            How to Train Your Dragon: The Hidden World
          </p>,
          <img
            alt=""
            src="http://image.tmdb.org/t/p/w185//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg"
          />,
        ],
        "className": "movie-card",
        "data-test": "MovieCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "How to Train Your Dragon: The Hidden World",
          },
          "ref": null,
          "rendered": "How to Train Your Dragon: The Hidden World",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://image.tmdb.org/t/p/w185//xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
